crates/net/network/src/transactions.rs

```diff
@@ -1074,116 +1255,287 @@ struct Peer {
 ///
 /// This will keep track of unique transaction hashes that are currently being fetched and submits
 /// new requests on announced hashes.
-#[derive(Debug, Default)]
+#[derive(Debug)]
+#[pin_project]
 struct TransactionFetcher {
+    /// All peers to which a request for pooled transactions is currently active. Maps 1-1 to
+    /// `inflight_requests`.
+    active_peers: LruMap<PeerId, u8, ByLength>,
```

Comment:
This is a subset of active connections, right?
if we do proper cleanup on closed sessions:

https://github.com/paradigmxyz/reth/blob/5c2f9961768ebde29bdf51b2b7d75b7adc07e99e/crates/net/network/src/transactions.rs#L823-L823

then I think we don't need to cache old peers,

though I realize, that this might complicate the peerid fallback handling

Reply:
yes `transaction_fetcher.active_peers` is a subset of `peers`, it maps 1-1 one with `inflight_request`, I added that to the docs for `active_peers`. so every time a `GetPooledTxRequestFut` resolves, the respective peers is removed from `active_peers`. `active_peers`, refers to peers that are active in the transaction fetcher context.

Reply:
https://github.com/paradigmxyz/reth/blob/5c2f9961768ebde29bdf51b2b7d75b7adc07e99e/crates/net/network/src/transactions.rs#L1652-L1654

Reply:
as for the fallback logic, I decided to remove peers as fallback peers on op for efficiency. since fallback peers are stored in limited lru cache anyway, the list will stay in bounds.
https://github.com/paradigmxyz/reth/blob/5c2f9961768ebde29bdf51b2b7d75b7adc07e99e/crates/net/network/src/transactions.rs#L703-L707
called here where ended sessions are also removed from fallback peers
https://github.com/paradigmxyz/reth/blob/5c2f9961768ebde29bdf51b2b7d75b7adc07e99e/crates/net/network/src/transactions.rs#L724-L727

I will add a check to `filter_unseen_hashes` too though, to make sure we don't evict the oldest but active peer in favour of a newer but inactive peer.